### PR TITLE
Security fixes

### PR DESCRIPTION
**Pull request by jdfox**
# Motivation and Context
Adding Talisman and CSRF protection per "Implement security fixes" story

# What has changed
- Added Talisman and CSP policy to frontstage
- CSRF already implemented in frontstage using flask_wtf.csrf
- Adding Talisman and CSP policy to response-ops-ui (TBD)

# How to test?
- build frontstage container on new branch
- make test in response-ops
- spin up ras-rm-docker-dev
- stop response-ops container from master branch, and swap in the new one
- log into response-ops and verify correct headers in browser dev tools
- make acceptance_tests in rasrm-acceptance-tests

# Links
- https://trello.com/c/5DNdOXdt

# Screenshots (if appropriate):

### CODE REVIEW

File: response_operations_ui/views/errors.py [hunk 1]
```diff
@@ -1,6 +1,8 @@
 import logging
 
 from flask import Blueprint, flash, redirect, render_template, url_for, request
+# TODO: implement customised error responses for failed CSRF validation
+# from flask_wtf.csrf import CSRFError
```
Comment (warrenbailey) [inline response_operations_ui/views/errors.py:5]: do we need this?

File: response_operations_ui/__init__.py [hunk 2]
```diff
@@ -105,5 +132,7 @@ def user_loader(user_id):
     Session(app)
 
     setup_blueprints(app)
+    csrf.exempt(reporting_unit_bp)
+    csrf.exempt(messages_bp)
```
Comment (warrenbailey) [inline response_operations_ui/__init__.py:136]: why are these pages exempt?

File: response_operations_ui/__init__.py [hunk 3]
```diff
@@ -32,7 +48,18 @@ def __call__(self, environ, start_response):
...
 def create_app(config_name=None):
+    csp_policy = copy.deepcopy(CSP_POLICY)
     app = Flask(__name__)
+    csrf = CSRFProtect(app)
+    csrf.init_app(app)
+    Talisman(
+        app,
+        content_security_policy=csp_policy,
+        content_security_policy_nonce_in=['script-src'],
+        force_https=False,  # this is handled at the firewall
```
Comment (warrenbailey) [inline response_operations_ui/__init__.py:59]: that comment isn't technically correct, the load balancer is handling it

File: response_operations_ui/__init__.py [hunk 4]
```diff
@@ -6,18 +7,33 @@
...
 from response_operations_ui.logger_config import logger_initial_config
 from response_operations_ui.user import User
-from response_operations_ui.views import setup_blueprints
+from response_operations_ui.views import setup_blueprints, reporting_unit_bp, messages_bp
 from response_operations_ui.common.jinja_filters import filter_blueprint
 
 cf = ONSCloudFoundry()
 
+# TODO: review https://content-security-policy.com/, remove this comment if we're covered.
+CSP_POLICY = {
```
Comment (warrenbailey) [inline response_operations_ui/__init__.py:25]: It looks ok to me, can we remove the comment?

File: response_operations_ui/__init__.py (hunk 3 above)
Comment (jdfox) [reply response_operations_ui/__init__.py]: corrected

File: response_operations_ui/__init__.py [hunk 5]
```diff
@@ -32,7 +47,18 @@ def __call__(self, environ, start_response):
 
 
 def create_app(config_name=None):
+    csp_policy = copy.deepcopy(CSP_POLICY)
     app = Flask(__name__)
+    csrf = CSRFProtect(app)
+    csrf.init_app(app)
```
Comment (warrenbailey) [inline response_operations_ui/__init__.py:53]: This isn't needed either

File: response_operations_ui/__init__.py (hunk 2 above)
Comment (jdfox) [reply response_operations_ui/__init__.py]: Let's discuss offline how to smoke test without these exemptions.

**Pull request merged**